package login;

enum error {
    success = 0;
    auth = 1;
    freq = 2;
	ban = 3;
	version = 4;
    unknow = 99;
}

// client -> login
message login_request {
    required string account = 1;
    required string passwd = 2;
	required int32 version = 3;
}

// login -> center
message user_login_request {
    required int32 tempid = 1;
    required uint64 uid = 2;
}

// center -> gate
message user_session_request {
    required int32 tempid = 1;
    required uint64 uid = 2;
    required string sk = 3;
}

// gate -> center
message user_session_reply {
    required error err = 1 [default = success];
    required int32 tempid = 2;
    optional uint64 uid = 3;
    optional string sk = 4;
}

// center -> login
message user_login_reply {
    required error err = 1 [default = success];
    required int32 tempid = 2;
    optional uint64 uid = 3;
    optional string sk = 4;
    optional string gateip = 5;
    optional int32 gateport = 6;
}

// login -> client
message login_reply {
    required error err = 1 [default = success];
    optional uint64 uid = 2;
    optional string sk = 3;
    optional string gateip = 4;
    optional int32 gateport = 5;
}

// client -> gate 
message connect_request {
    required uint64 uid = 1;
    required string sk = 2;
};

// gate -> client
message connect_reply {
    required error err = 1;
};

// center -> login
message center_reg {
    required int32 id = 1;
}

// gate -> center
message gate_reg {
    required string ip = 1;
    required int32 port = 2;
}
